packages/core/echo/echo-pipeline/src/automerge/automerge-host.ts

```diff
@@ -327,12 +364,121 @@ export class AutomergeHost extends Resource {
     const handle = this._repo.handles[documentId];
     if (handle) {
       const doc = handle.docSync();
-      if (!doc) {
-        return undefined;
+      if (doc) {
+        return getHeads(doc);
+      }
+    }
+    return this._headsStore.getHeads(documentId);
+  }
+
+  //
+  // Collection sync.
+  //
+
+  getLocalCollectionState(collectionId: string): CollectionState | undefined {
```

Comment:
not sure if these wrapper methods are useful, just adds an extra hop during navigation to the actual logic

Reply:
I thought about it too, but this seems like a lesser evil then exposing the internals -- in the future it will be much easier to swap out the implementation. Also if you think about the architecture as a bunch of layered components, then wrapper methods help make the API explicit at every layer. Of course none of this is universal, and sometimes it makes sense to expose a getter to a component (e.g. `space.db`), but I think in this case I'd prefer to hide the collection synchronizer

Reply:
I also want to fully hide the Repo inside AutomergeHost, hence the methods for creating and querying documents on this class

Reply:
>  in the future it will be much easier to swap out the implementation

you can always add a getter with limited-API interface return type. but not that I'm insisting